elasticsearch/es_client_v7.go

```diff
@@ -381,3 +381,109 @@ func (es *ESClientV7) CreateDBUserRole(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (es *ESClientV7) IndexExistsOrNot(_index string) (bool, error) {
+	req := esapi.IndicesExistsRequest{
+		Index: []string{_index},
+	}
+	res, err := req.Do(context.Background(), es.client)
+	if err != nil {
+		return false, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode > 299 {
+		return false, nil
```

Comment:
if res.IsError() {
		return errors.New(fmt.Sprintf("Failed to get response from index exist request with error statuscode %d", res.StatusCode))
	}